state/migration_import.go

```diff
@@ -1174,6 +1164,36 @@ func (i *importer) unit(s description.Application, u description.Unit, ctrlCfg c
 	return nil
 }
 
+func (i *importer) importUnitState(unit *Unit, u description.Unit, ctrlCfg controller.Config) error {
+	us := NewUnitState()
+
+	if charmState := u.CharmState(); len(charmState) != 0 {
+		us.SetCharmState(charmState)
+	}
+	if relationState := u.RelationState(); len(relationState) != 0 {
+		us.SetRelationState(relationState)
+	}
+	if uniterState := u.UniterState(); uniterState != "" {
+		us.SetUniterState(uniterState)
+	}
+	if storageState := u.StorageState(); storageState != "" {
+		us.SetStorageState(storageState)
+	}
+	if meterStatusState := u.MeterStatusState(); meterStatusState != "" {
+		us.SetMeterStatusState(meterStatusState)
+	}
+
+	// No state to persist.
+	if !us.Modified() {
```

Comment:
Can we test this scenario?

Reply:
What would be the purpose of this test? That we don't call SetState if the description is empty or that `us.Modified()` works as expected (the latter is already tested in state)

Reply:
I want to see that we don't call SetState :)